enterprise/backend/src/metabase_enterprise/workspaces/common.clj

```diff
@@ -0,0 +1,29 @@
+(ns metabase-enterprise.workspaces.common
+  (:require
+   [metabase.api-keys.core :as api-key]
+   [metabase.util.log :as log]
+   [metabase.util.malli :as mu]
+   [metabase.util.secret :as u.secret]
+   [toucan2.core :as t2]))
+
+(mu/defn create-workspace
+  "Create workspace"
+  [user-id :- pos-int?
+   {:keys [name graph]} :- [:map
+                            [:name :string]
+                            [:graph [:map]]]]
+  (let [api-key (api-key/create-api-key-with-new-user!
+                 {:key-name (format "API key for Workspace %s" name)
+                  :key-scope :api-key.scope/workspace})
+        ws      (t2/insert-returning-instance! :model/Workspace
+                                               {:name       name
+                                                :graph      graph
+                                                :user_id    user-id
```

Comment:
There are two types of user for a workspace:

- The creator / owner
- The restricted user associated with the api-key, which can only work inside the workspace collection and uses the workspace db_user instead of the regular connection.

This seems to conflate them.

I suggest for now that we rename this to `creator_id` for clarity, and defer the `execution_user` and its api-key stuff until later.

Reply:
Ah I see the api key _is_ using a new user. If you're going to to create that now let's make sure to save the user_id against the workspace too, as it'll be very useful, for example as the executor when the author / another admin performs execution.